prefect_kubernetes/pods.py

```diff
@@ -0,0 +1,361 @@
+    namespace: Optional[str] = "default",
+    **kube_kwargs: Dict[str, Any],
+) -> V1Pod:
+    """Delete a Kubernetes pod in a given namespace.
+
+    Args:
+        kubernetes_credentials: `KubernetesCredentials` block for creating
+            authenticated Kubernetes API clients.
+        pod_name: The name of the pod to delete.
+        body: A Kubernetes `V1DeleteOptions` object.
+        namespace: The Kubernetes namespace to delete this pod from.
+        **kube_kwargs: Optional extra keyword arguments to pass to the Kubernetes API.
+
+    Returns:
+        A Kubernetes `V1Pod` object.
+
+    Example:
+        Delete a pod in the default namespace:
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.pods import delete_namespaced_pod
+        from kubernetes.client.models import V1DeleteOptions
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_pod_metadata = delete_namespaced_pod(
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+                pod_name="test-pod",
+                body=V1DeleteOptions(grace_period_seconds=0),
+            )
+        ```
+    """
+    with kubernetes_credentials.get_client("core") as core_v1_client:
+
+        return await run_sync_in_worker_thread(
+            core_v1_client.delete_namespaced_pod,
+            pod_name,
+            body=body,
+            namespace=namespace,
+            **kube_kwargs,
+        )
+
+
+@task
+async def list_namespaced_pod(
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    **kube_kwargs: Dict[str, Any],
+) -> V1PodList:
+    """List all pods in a given namespace.
+
+    Args:
+        kubernetes_credentials: `KubernetesCredentials` block for creating
+            authenticated Kubernetes API clients.
+        namespace: The Kubernetes namespace to list pods from.
+        **kube_kwargs: Optional extra keyword arguments to pass to the Kubernetes API.
+
+    Returns:
+        A Kubernetes `V1PodList` object.
+
+    Example:
+        List all pods in the default namespace:
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.pods import list_namespaced_pod
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_pod_list = list_namespaced_pod(
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds")
+            )
+        ```
+    """
+    with kubernetes_credentials.get_client("core") as core_v1_client:
+
+        return await run_sync_in_worker_thread(
+            core_v1_client.list_namespaced_pod, namespace=namespace, **kube_kwargs
+        )
+
+
+@task
+async def patch_namespaced_pod(
+    kubernetes_credentials: KubernetesCredentials,
+    pod_name: str,
+    body: V1Pod,
```

Comment:
Same comment about a clearer parameter name.
```suggestion
   pod_updates: V1Pod,
```